lib/blemulator/README.md

```diff
@@ -0,0 +1,253 @@
+            convenienceName: convenienceName);
+
+  @override
+  Future<void> write(Uint8List value) {
+    int valueAsInt = value.buffer.asByteData().getUint8(0);
+    if (valueAsInt != 0 && valueAsInt != 1) {
+      return Future.error(SimulatedBleError(
+          BleErrorCode.CharacteristicWriteFailed, "Unsupported value"));
+    } else {
+      return super.write(value); //this propagates value through the blemulator,
+      // allowing you to react to changes done to this characteristic
+    }
+  }
+}
+```
+
+
+#### Publishing notifications
+This time peripheral has additional logic. _IR Temperature Config_ characteristic turns the IR thermometer on (1) and off (0).
+ Regardless of that the _IR Temperature Data_ emits data each (_IR Temperature Period_ value times 10) milliseconds (as per specification linked above) and that may be actual reading or zero.
+
+```dart
+class SensorTag extends SimulatedPeripheral {
+  static const String _serviceUuid = "F000AA00-0451-4000-B000-000000000000";
+  static const String _temperatureDataUuid =
+      "F000AA01-0451-4000-B000-000000000000";
+  static const String _temperatureConfigUuid =
+      "F000AA02-0451-4000-B000-000000000000";
+  static const String _temperaturePeriodUuid =
+      "F000AA03-0451-4000-B000-000000000000";
+
+  bool _readingTemperature;
+
+  SensorTag(
+      {String id = "4B:99:4C:34:DE:77",
+      String name = "SensorTag",
+      String localName = "SensorTag"})
+      : super(
+            name: name,
+            id: id,
+            advertisementInterval: Duration(milliseconds: 800),
+            services: [
+              SimulatedService(
+                  uuid: _serviceUuid,
+                  isAdvertised: true,
+                  characteristics: [
+                    SimulatedCharacteristic(
+                      uuid: _temperatureDataUuid,
+                      value: Uint8List.fromList([0, 0, 0, 0]),
+                      convenienceName: "IR Temperature Data",
+                      isWritableWithoutResponse: false,
+                      isWritableWithResponse: false,
+                    ),
+                    SimulatedCharacteristic(
+                        uuid: _temperatureConfigUuid,
+                        value: Uint8List.fromList([0]),
+                        convenienceName: "IR Temperature Config"),
+                    SimulatedCharacteristic(
+                        uuid: _temperaturePeriodUuid,
+                        value: Uint8List.fromList([50]),
+                        convenienceName: "IR Temperature Period"),
+                  ],
+                  convenienceName: "Temperature service"),
+            ]) {
+    scanInfo.localName = localName;
+
+    getCharacteristicForService(_serviceUuid, _temperatureConfigUuid)
+        .monitor()
+        .listen((value) {
+      int valueAsInt = value.buffer.asByteData().getUint8(0);
+      _readingTemperature = valueAsInt == 1 ? true : false;
+    });
+
+    _emitTemperature();
```

Comment:
Will this practically launch a new thread that will emit values?

Reply:
In a way. While this is the same thread, it runs like a saga in RN. Since the `_emitTemperature()` function is marked as async and there's no `await` keyword before this call, it won't pause the execution of this method.

There are isolates in Dart that are physical threads, but they cannot share any context with the main thread.